src/main/java/decoder/Decoder.java

```diff
@@ -0,0 +1,25 @@
+package decoder;
+
+import com.google.protobuf.util.JsonFormat;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import proto.DataFormats.SerializedData;
+
+/** Decode data from the proto format to JSON. */
+public class Decoder {
+  public static void decode(InputStream dataStream, OutputStream outputStream) {
+
+    // At some point it would make sense to have this data read in chunks of the
+    // data file and include headers in the proto definitions so we could
+    // address issues with large files, stuff not fitting into RAM, etc.
+    try {
+      SerializedData protoData = SerializedData.parseFrom(dataStream);
```

Comment:
I think here your goal is decode one record each time instead of the whole file at one time(some data will be very big that you are not able to store them all in memory), 
then you may need Frelica provide you an stream writer like DatabaseWriter->WriteProto(timestamp, PoseProto);  so you can write each piece of data into db.

Reply:
when you say one record, are you talking about decoding 1 point's location and orientation?

Reply:
Yes, each pose also has a timestamp, so one record is a pair of <timestamp, pose>.